cartographer_ros/cartographer_ros/node.cc

```diff
@@ -231,8 +230,10 @@ void Node::PublishTrajectoryStates(const ::ros::WallTimerEvent& timer_event) {
 void Node::PublishTrajectoryNodesList(
     const ::ros::WallTimerEvent& unused_timer_event) {
   carto::common::MutexLocker lock(&mutex_);
-  trajectory_nodes_list_publisher_.publish(
-      map_builder_bridge_.GetTrajectoryNodesList());
+  if (trajectory_nodes_list_publisher_.getNumSubscribers() > 0) {
```

Comment:
I have, the impact wasn't big. I measured offline node's time to crunch a 450s portion of a Deutsches museum bag (without the final optimization step, 20 measurements). With publishing the trajectory: 107.5360 seconds ± 5.1576. Without publishing the trajectory: 101.2620 seconds ± 4.2812 seconds. Not a huge difference, but still, I think a single if statement is worth the small savings.